### PR TITLE
support SCA packages in CycloneDX output

### PR DESCRIPTION
**Pull request by gruebel**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

- added support for extra resources/packages for SCA package reports
- added support for SCA package components and CVE vulnerabilities in CycloneDX output

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/8.Outputs/CycloneDX BOM.md [hunk 1]
```diff
@@ -206,3 +262,42 @@ ex.
...
+  <name>flask</name>
+  <version>0.6</version>
+  <purl>pkg:pypi/flask@0.6</purl>
+</component>
+```
+
+- `group`: Group name of the package
+- `name`: Name of the package
+- `version`: Version of the package
+- `purl`: Format `pkg:<package type>/<package group name>/<packge name>@<package version>`
```
Comment (ChanochShayner) [inline docs/8.Outputs/CycloneDX BOM.md:303]: Awesome, informative, and clear.

**Review by ChanochShayner — APPROVED** [review]

File: docs/8.Outputs/CycloneDX BOM.md (hunk 1 above)
Comment (ChanochShayner) [reply docs/8.Outputs/CycloneDX BOM.md:303]: Do we want to add the repo name or ID?
for a case that we are merging two repos reports.

**Comment by ChanochShayner** [comment]
Another question - 
It seems that you saved all the materials in the same report file, but in CSV format we save a report file per material.
WDYT?
@PelegLi FYI

**Comment by gruebel** [comment]
> Another question - It seems that you saved all the materials in the same report file, but in CSV format we save a report file per material. WDYT? @PelegLi FYI

for a `checkov` run it makes more sense to add everything in one file. The code itself can be used to create separate reports for each material.

**Review by PelegLi — APPROVED** [review]
💣

**Pull request merged**